### ISSUE TITLE
Long File Names

### ISSUE THREAD

**Issue by willtebbutt** [tkluck/StatProfilerHTML.jl#27]
Using Julia 1.7.1 and StatProfilerHTML ~~1.3~~ 1.1, but the same problem occus on 1.3, I get the following error after running either `statprofilehtml` on the output from `Profile.@profile` or `@profilehtml`:
```julia
ERROR: SystemError: opening file "/Users/willtebbutt/ml/downscaling-gcm-combinations/environments/hadiod/statprof/gaussian_process_probabilistic_programme.jl-L1VzZXJzL3dpbGx0ZWJidXR0L21sL2Rvd25zY2FsaW5nLWdjbS1jb21iaW5hdGlvbnMvZW52aXJvbm1lbnRzL2hhZGlvZC9kZXYvSW5maW5pdGVMaW5lYXJNaXhpbmdNb2RlbHMvZGV2L1N0aGVuby9zcmMvZ2F1c3NpYW5fcHJvY2Vzc19wcm9iYWJpbGlzdGljX3Byb2dyYW1tZS5qbA==.html": File name too long
Stacktrace:
  [1] systemerror(p::String, errno::Int32; extrainfo::Nothing)
    @ Base ./error.jl:174
  [2] #systemerror#68
    @ ./error.jl:173 [inlined]
  [3] systemerror
    @ ./error.jl:173 [inlined]
  [4] open(fname::String; lock::Bool, read::Nothing, write::Nothing, create::Nothing, truncate::Bool, append::Nothing)
    @ Base ./iostream.jl:293
  [5] open(fname::String, mode::String; lock::Bool)
    @ Base ./iostream.jl:355
  [6] open(fname::String, mode::String)
    @ Base ./iostream.jl:355
  [7] open(::StatProfilerHTML.HTML.var"#99#105"{StatProfilerHTML.Reports.Report, Symbol, Vector{Tuple{LineNumberNode, String}}}, ::String, ::Vararg{String}; kwargs::Base.Pairs{Symbol, Union{}, Tuple{}, NamedTuple{(), Tuple{}}})
    @ Base ./io.jl:328
  [8] open(::Function, ::String, ::String)
    @ Base ./io.jl:328
  [9] output(r::StatProfilerHTML.Reports.Report, path::String)
    @ StatProfilerHTML.HTML ~/.julia/packages/StatProfilerHTML/BuyY0/src/HTML.jl:68
 [10] statprofilehtml(data::Vector{UInt64}, litrace::Dict{UInt64, Vector{Base.StackTraces.StackFrame}}; from_c::Bool)
    @ StatProfilerHTML ~/.julia/packages/StatProfilerHTML/BuyY0/src/StatProfilerHTML.jl:23
 [11] statprofilehtml (repeats 3 times)
    @ ~/.julia/packages/StatProfilerHTML/BuyY0/src/StatProfilerHTML.jl:17 [inlined]
 [12] macro expansion
    @ ~/.julia/packages/StatProfilerHTML/BuyY0/src/StatProfilerHTML.jl:37 [inlined]
 [13] optimise_alt_approx(<really_long_type>)
    @ Main ./REPL[56]:45
 [14] top-level scope
    @ REPL[88]:1
```
Is there a good way to avoid this problem?

**Comment by willtebbutt** [comment]
This is interesting, because it appears that this file also doesn't exist in my `/statprof` folder.

**Comment by tkluck** [comment]
Thanks for the report, @willtebbutt . What's happening here is that StatProfilerHTML generates an output filename from a source file's basename and the encoded full path: https://github.com/tkluck/StatProfilerHTML.jl/blob/master/src/HTML.jl#L22. I didn't want to have to reproduce a full `/home/....` or `/usr/...` as some subdirectories of the `statprof` directory, so I ended up using a flat directory structure and using the base64 encoding to avoid name clashes. This causes the problem you're seeing, because the path can be of arbitrary length but obviously the file name has a limit.

An easy way to solve it is to just use a hash instead of an encoding. I'll publish a fix.

**Comment by tkluck** [comment]
This should be fixed in 1.4, which releases soon (https://github.com/JuliaRegistries/General/pull/51302).

**Comment by willtebbutt** [comment]
Fantastic. Thanks for sorting this out so quickly, and thanks for the excellent package!
